drift-server/README.md

```diff
@@ -0,0 +1,84 @@
+# Drift Server
+
+Drift Server is a simple library for creating Thrift services using annotations.
+
+## Implementing a Service
+
+In Drift, a Thrift service is simply a Java class annotated with `@ThriftService`. In
+addition to annotating the class directly, Drift supports annotations on super classes
+or interfaces. A single service instance is used for all incoming client requests, so
+it must be thread safe and should handle concurrent requests. For example, the
+following describes a Scribe service:
+
+```java
+@ThriftService
+public class Scribe
+{
+    @ThriftMethod
+    public ResultCode log(List<LogEntry> messages)
+    {
+        // do something with messages
+        return ResultCode.OK;
+    }
+}
+```
+
+To make the service method asynchronous, simply change the return type of the
+method to `ListenableFuture`:
+
+```java
+@ThriftService
+public class Scribe
+{
+    public ListenableFuture<ResultCode> log(List<LogEntry> messages)
+    {
+        // do something with messages, then complete the future
+        return immediateFuture(ResultCode.OK);
+    }
+}
+```
+
+The future should be completed with the value to be returned to the client.
+
+### Parameters and Exceptions
+
+See the documentation for [Drift Client](../drift-client).
+
+## Creating a Server
+
+The following code manually constructs a `DriftServer` using the Netty transport:
+
+```java
+// service handler (must be thread safe)
+Scribe service = new Scribe();
+
+// create the server
+DriftServer driftServer = new DriftServer(
+        new DriftNettyServerTransportFactory(config),
+        new ThriftCodecManager(),
+        new NullMethodInvocationStatsFactory(),
+        ImmutableSet.of(service),
```

Comment:
maybe we should note somewhere that multiple services can be bound as long as the method names are unique (a Thrift restriction)